app/main.py

```diff
@@ -149,20 +150,28 @@ def collect_item():
     Collect a tap and forward it on to XOS with the label ID.
     """
     has_tapped = HasTapped.get_or_none(has_tapped=0)
-    if has_tapped:
-        has_tapped.has_tapped = 1
-        has_tapped.save()
+    if not has_tapped:
+        return HTTPError('Tap still processing.'), abort(500, 'Tap still processing.')
+    has_tapped.tap_successful = 0
+    has_tapped.has_tapped = 1
+
     xos_tap = dict(request.get_json())
     try:
         record = model_to_dict(Label.select().order_by(Label.datetime.desc()).get())
     except DoesNotExist:
+        has_tapped.save()
```

Comment:
Separating the `.save()` from where you set the data doesn't feel right. What was your logic behind doing this?

Reply:
As discussed, explicitly set data where saving. Along with `tap_processing` the logic is all a bit more readable now.